oak_attestation_verification/src/verifier.rs

```diff
@@ -465,6 +462,9 @@ fn verify_root_layer(
     endorsements: Option<&RootLayerEndorsements>,
     reference_values: &RootLayerReferenceValues,
 ) -> anyhow::Result<()> {
+    if let Some(insecure_values) = reference_values.insecure.as_ref() {
+        return verify_insecure(insecure_values).context("insecure root layer verification failed");
+    }
```

Comment:
I believe this was the suggested approach from the discussion in the sync meeting yesterday. We need a mechanism to skip verifying the signature of the attestation report. Conceptually it also makes sense, since "insecure" verification shouldn't fail if the workload is running on a more secure platform. There is a risk though that someone adds insecure and AMD reference values, and the verification will always pass even with an invalid AMD VCEK certificate so we have to make sure the implications is very clear in the documentation.

I agree with the point that we should clarify this usage in the comments of the InsecureReferenceValues.

Reply:
Agree with expanding the comment. I'm also wondering if there is a good way to log our skips on the client, to make them more visible? Not sure if we can log it directly in the Rust code so that logs are visible both on Android and when running on the server. But maybe we could add a "warnings" field to the AttestationResults? Then it would be easy to log them in OakClient

Reply:
Recording the skips and returning them as part of AttestationResults sounds good.

Regarding the InsecureReferenceValues, in my understanding it is a collection of all reference values which should be verified when no specialized hardware is available, it just happens to be empty. So "insecure" is a platform value, not a switch. There are two switches of interest:

1. Skipping verification for a particular platform value - IMO use skip {} for that
2. Rejecting a platform value - e.g. making verification fail if no specialized hardware 

I don't intend to push back though, would be fine with extensive comments.

Reply:
Rearranged the check to make sure that if matching TEE reference values are specified they are verified in any case, though this undoes addressing Tiziano's comment. WDYT?